source/adapters/level_zero/command_buffer.cpp

```diff
@@ -180,16 +181,17 @@ ur_result_t calculateKernelWorkDimensions(
           --GroupSize[I];
         }
         if (GlobalWorkSize[I] / GroupSize[I] > UINT32_MAX) {
-          urPrint("urCommandBufferAppendKernelLaunchExp: can't find a WG size "
-                  "suitable for global work size > UINT32_MAX\n");
+          logger::debug(
+              "urCommandBufferAppendKernelLaunchExp: can't find a WG size "
+              "suitable for global work size > UINT32_MAX");
           return UR_RESULT_ERROR_INVALID_WORK_GROUP_SIZE;
         }
         WG[I] = GroupSize[I];
       }
-      urPrint(
-          "urCommandBufferAppendKernelLaunchExp: using computed WG size = {%d, "
-          "%d, %d}\n",
-          WG[0], WG[1], WG[2]);
+      logger::debug("urCommandBufferAppendKernelLaunchExp: using computed WG "
+                    "size = {{{}, "
```

Comment:
it seems like you'd have an extra pair of `{` - is this expected?

actually, general question - have you tested all these works? 😄

Reply:
Double {{ means a single '{'

Reply:
I see, thx. Nonetheless, can you paste a few example outputs from these logs? 😉

Reply:
```
<LOADER>[DEBUG]: Logger loader initialized successfully!
Platform initialized.
<LEVEL_ZERO>[DEBUG]: ZE ---> zeInit(ZE_INIT_FLAG_GPU_ONLY)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGet(&ZeDriverCount, nullptr)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGet(&ZeDriverCount, ZeDrivers.data())
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetProperties(ZeDriver, &ZeDriverProperties)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetApiVersion(ZeDriver, &ZeApiVersion)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetExtensionProperties(ZeDriver, &Count, nullptr)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetExtensionProperties(ZeDriver, &Count, ZeExtensions.data())
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetExtensionFunctionAddress(DriverHandle, "zexDriverImportExternalPointer", reinterpret_cast<void **>(&zexDriverImportExternalPointer))
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDriverGetExtensionFunctionAddress(DriverHandle, "zexDriverReleaseImportedPointer", reinterpret_cast<void **>(&zexDriverReleaseImportedPointer))
API version: 0.7
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGet(ZeDriver, &ZeDeviceCount, nullptr)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGet(ZeDriver, &ZeDeviceCount, ZeDevices.data())
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGetProperties(ZeDevice, &Properties)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGetCommandQueueGroupProperties(ZeDevice, &numQueueGroups, nullptr)
<LEVEL_ZERO>[INFO]: Number of queue groups = 1
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGetCommandQueueGroupProperties(ZeDevice, &numQueueGroups, QueueGroupProperties.data())
<LEVEL_ZERO>[INFO]: main blitter/copy engine is not available
<LEVEL_ZERO>[INFO]: link blitter/copy engines are not available
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGetSubDevices(Device->ZeDevice, &SubDevicesCount, nullptr)
<LEVEL_ZERO>[DEBUG]: ZE ---> zeDeviceGetSubDevices(Device->ZeDevice, &SubDevicesCount, ZeSubd
```